corehq/apps/dump_reload/couch/dump.py

```diff
@@ -1,56 +1,51 @@
-import itertools
 import json
 from collections import Counter
 
 from couchdbkit import ResourceNotFound
 
-from dimagi.utils.chunked import chunked
-from dimagi.utils.couch.bulk import get_docs
-from dimagi.utils.couch.database import iter_docs
-
 from corehq.apps.dump_reload.couch.id_providers import (
     DocTypeIDProvider,
     DomainInListKeyGenerator,
     DomainKeyGenerator,
-    UserIDProvider,
+    WebUserIDProvider,
     ViewIDProvider,
 )
 from corehq.apps.dump_reload.exceptions import DomainDumpError
 from corehq.apps.dump_reload.interface import DataDumper
 from corehq.feature_previews import all_previews
+from dimagi.utils.couch.database import iter_docs
 
 DOC_PROVIDERS = {
-    DocTypeIDProvider(['Application']),
-    DocTypeIDProvider(['CommtrackConfig']),
+    DocTypeIDProvider('Application'),
+    DocTypeIDProvider('CommtrackConfig'),
     ViewIDProvider('CommCareMultimedia', 'hqmedia/by_domain', DomainKeyGenerator()),
-    DocTypeIDProvider(['MobileAuthKeyRecord']),
-    DocTypeIDProvider(['Product']),
-    DocTypeIDProvider(['Program']),
-    UserIDProvider(include_mobile_users=False),
-    DocTypeIDProvider(['CommCareUser']),
-    DocTypeIDProvider(['UserRole']),
-    DocTypeIDProvider(['Group']),
-    DocTypeIDProvider(['ReportConfiguration']),
-    DocTypeIDProvider(['ReportNotification']),
-    DocTypeIDProvider(['ReportConfig']),
-    DocTypeIDProvider(['DataSourceConfiguration']),
-    DocTypeIDProvider(['FormExportInstance']),
-    DocTypeIDProvider(['FormExportDataSchema']),
-    DocTypeIDProvider(['ExportInstance']),
-    DocTypeIDProvider(['ExportDataSchema']),
-    DocTypeIDProvider(['CaseExportInstance']),
-    DocTypeIDProvider(['CaseExportDataSchema']),
-    DocTypeIDProvider(['FixtureOwnership']),
-    DocTypeIDProvider(['FixtureDataType']),
-    DocTypeIDProvider(['FixtureDataItem']),
+    DocTypeIDProvider('MobileAuthKeyRecord'),
+    DocTypeIDProvider('Product'),
+    DocTypeIDProvider('Program'),
+    WebUserIDProvider(),
```

Comment:
I like what you've done with the class's name, and that the class uses the view intended for retrieving WebUsers.

Did you consider dropping the class completely, and using `DocTypeIDProvider('WebUser')`, as "CommCareUser" does on the next line?

It does make sense to use `get_all_user_ids_by_domain()` for retrieving users instead of `get_doc_ids_in_domain_by_type()` ... but on the other hand I can also see why we wouldn't want a different class just for CommCareUsers when `DocTypeIDProvider` can do the job. It seems we went with the first approach with `UserIDProvider()` and the second approach with `DocTypeIDProvider('CommCareUser')`, and it's not obvious to me why we chose not to be consistent.

Reply:
Web users can't use the `DocTypeIDProvider` because they can belong to multiple domains so we need to query a different view. Having separate classes makes sense from a complexity point of view (keeping the complexity of the 'UserIDProvider' class low and reusing an existing class for CommCareUsers.